keras_cv/layers/preprocessing/cut_mix.py

```diff
@@ -0,0 +1,142 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+from tensorflow.keras import layers
+
+
+class CutMix(layers.Layer):
+    """
+    CutMix implements the CutMix data augmentation technique as proposed in https://arxiv.org/abs/1905.04899.
+
+    Args:
+        num_classes: number of classes in the dataset.
```

Comment:
Just curious why the total number of class is needed for this class. If this is only needed for label smoothing, should it be an optional param? eg only specify this when label smoothing is not 0.

Reply:
I ended up actually entirely removing num_classes, so even better